### PR TITLE
Remove/replace more py2 compat code

### PR DESCRIPTION
**Pull request by rahulporuri**
- string_types on python 3 is just str
- unicode_type on python 3 is just str
- iteritems on python 3 is just dict.items()
- remove 2/3 conditionals

### CODE REVIEW

**Comment by ccordoba12** [comment]
@rahulporuri, thanks for helping us to move this package to Python 3.

@dalthviz, please take a look at this one.

File: qtconsole/util.py [hunk 1]
```diff
@@ -25,7 +24,7 @@ class MetaQObjectHasTraits(MetaQObject, MetaHasTraits):
     def __new__(mcls, name, bases, classdict):
         # FIXME: this duplicates the code from MetaHasTraits.
         # I don't think a super() call will help me here.
-        for k,v in iteritems(classdict):
+        for k,v in classdict.items():
```
Comment (ccordoba12) [inline qtconsole/util.py:27]: This previously returned an iterator:

https://github.com/ipython/ipython_genutils/blob/7daa8b9296038f82cca7376a4e044673a115e94f/ipython_genutils/py3compat.py#L188

Please leave things as they were before.

**Review by ccordoba12 — COMMENTED** [review]
@rahulporuri, an small comment for you. Otherwise looks good.

File: qtconsole/util.py (hunk 1 above)
Comment (rahulporuri) [reply qtconsole/util.py]: changes undone.

File: qtconsole/util.py [hunk 2]
```diff
@@ -24,7 +25,7 @@ class MetaQObjectHasTraits(MetaQObject, MetaHasTraits):
     def __new__(mcls, name, bases, classdict):
         # FIXME: this duplicates the code from MetaHasTraits.
         # I don't think a super() call will help me here.
-        for k,v in classdict.items():
+        for k,v in iteritems(classdict):
```
Comment (ccordoba12) [inline qtconsole/util.py]: Sorry, I think you didn't understand me. Changing this to

    for k,v in iter(classdict.items()):

is enough. That way we can remove the dependency in `ipython_genutils`.

Comment (rahulporuri) [reply qtconsole/util.py]: ha. fixed again.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me, thanks @rahulporuri!

**Pull request merged**